### DISCUSSION TITLE
argo-workflows/walk-through/sidecars/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9431] [General]
# argo-workflows/walk-through/sidecars/

https://argoproj.github.io/argo-workflows/walk-through/sidecars/

**Comment by apiwoni** [comment]
If I defined sidecar before the container (main) would this guarantee at least start-up order as is done by default in Kubernetes because Kublet code is single threaded and starts up containers in order defined in spec.containers

**Comment by aixi** [comment]
when sidecar container fail, is there a way to only restart sidecar ?
